elpy.el

```diff
@@ -1737,6 +1737,84 @@ If ASK-FOR-EACH-ONE is non-nil, ask before killing each python process.
                         (- indent-level))
         (buffer-string)))))
 
+;;;;;;;;;;;;
+;;; Profiler
+
+(defcustom elpy-profile-visualizer "snakeviz"
+  "Visualizer for elpy profile results."
+  :type 'str)
+
+(defun elpy-profile-buffer-or-region (&optional in-dir dont-display)
+  "Profile asynchronously the active region or the current buffer
+and display the result using `elpy-profile-visualizer'.
+
+If IN-DIR is non nil, profile result is saved in the same
+directory as the script.
+If DONT-DISPLAY is non nil, don't display the profile results."
+  (interactive "P")
+  (let* ((file-name (buffer-name))
+         (file-dir (file-name-directory (buffer-file-name)))
+	 (beg (if (region-active-p) (region-beginning) (point-min)))
+	 (end (if (region-active-p) (region-end) (point-max)))
+	 (tmp-file-prefix (if (region-active-p) "_region_" ""))
+	 (tmp-file (if in-dir (concat file-dir "/" tmp-file-prefix file-name)
+                     (concat (make-temp-file "elpy-profile-" t)  "/" tmp-file-prefix file-name)))
+         (region (elpy-shell--region-without-indentation beg end)))
+    (with-temp-buffer
+      (insert region)
+      (write-region (point-min) (point-max) tmp-file nil t))
+    (elpy-profile--file tmp-file t dont-display) dont-display))
+
+(defun elpy-profile--file (file &optional in-dir dont-display)
+  "Profile asynchronously FILE and display the result using
+`elpy-profile-visualizer'.
+
+If IN-DIR is non nil, profile result is saved in the same
+directory as the script.
+If DONT-DISPLAY is non nil, don't display the profile results."
+  ;; clean log buffers
+  (condition-case nil
+      (with-current-buffer " *elpy-profile-log*"
+	(kill-this-buffer))
+    (error nil))
+  (condition-case nil
+      (with-current-buffer "*elpy-profile-log*"
+	(kill-this-buffer))
+    (error nil))
+  (let* ((prof-file (if in-dir (concat (file-name-sans-extension file) ".profile")
+		      (concat (make-temp-file "elpy-profile-" nil ".profile"))))
```

Comment:
Please trust Emacs to indent your code correctly. Also, add a newline after the condition to make this more readable.